cmd/query/main.go

```diff
@@ -57,6 +64,13 @@ func main() {
 			queryOpts := new(builder.QueryOptions).InitFromViper(v)
 			sFlags := new(flags.SharedFlags).InitFromViper(v)
 
+			var hcChannel = make(chan int)
+			go func() {
+				if _, err := healthcheck.Serve(hcChannel, http.StatusServiceUnavailable, queryOpts.QueryHealthCheckHTTPPort, logger); err != nil {
+					logger.Fatal("Could not start the health check server.", zap.Error(err))
+				}
+			}()
+
```

Comment:
I still have the issue with the verbosity of it. Most of the logic can be moved inside the healthChecker itself, e.g. main() does not need to know anything about the hcChannel. Literally, the usage can be reduced to this:

```go
healthServer, err := healthcheck.Serve(queryOpts.QueryHealthCheckHTTPPort, logger)
if err != nil {
    logger.Fatal("Could not start the health check server.", zap.Error(err))
}
```

and once everything is started

```go
healthServer.Ready()
```

the http status codes can also be hidden inside the implementation.